ui/src/app/views/workspace-library.tsx

```diff
@@ -0,0 +1,176 @@
+  },
+  menuLink: {
+    color: colors.accent, fontSize: 14, fontWeight: 600, display: 'flex',
+    flexDirection: 'row', alignItems: 'center', padding: '0.35rem 0', borderRadius: '3px'
+  },
+  divider: {
+    width: '100%', backgroundColor: colors.primary, borderWidth: '0px', height: '1px',
+    marginBottom: '0.5rem'
+  },
+  menuLinkSelected: {
+    backgroundColor: '#F3F2F7'
+  }
+});
+
+const libraryTabEnums = {
+  FEATURED_WORKSPACES: {
+    title: 'Featured Workspaces',
+    icon: 'star'
+  },
+  PUBLISHED_WORKSPACES: {
+    title: 'Published Workspaces',
+    icon: 'library'
+  }
+};
+
+const libraryTabs = [
+  libraryTabEnums.FEATURED_WORKSPACES,
+  libraryTabEnums.PUBLISHED_WORKSPACES
+];
+
+const LibraryTab: React.FunctionComponent<{
+  title: string, icon: string, onClick: Function, selected: boolean}> =
+  ({title, icon, onClick, selected}) => {
+    return <Clickable style={selected ? {...styles.menuLink, ...styles.menuLinkSelected}
+    : styles.menuLink}
+                      onClick={onClick} hover={styles.menuLinkSelected}>
+      <ClrIcon shape={icon} style={styles.iconStyling} class='is-solid' size={24}/>
+      {title}
+    </Clickable>;
+  };
+
+export const WorkspaceLibrary = withUserProfile()
+(class extends React.Component<
+  {profileState: { profile: Profile, reload: Function } },
+  {currentTab: {title: string, icon: string}, errorText: string,
+    workspaceList: WorkspacePermissions[], workspacesLoading: boolean}> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentTab: libraryTabEnums.PUBLISHED_WORKSPACES,
+      errorText: '',
+      workspaceList: [],
+      workspacesLoading: true
+    };
+  }
+
+  componentDidMount() {
+    this.reloadPublishedWorkspaces();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    // Reload libraries when switching tabs
+    if (this.state.currentTab !== prevState.currentTab) {
+      console.log('here');
+      this.updateListedWorkspaces(this.state.currentTab);
+    }
+  }
+
+  updateListedWorkspaces(tabEnum) {
+    switch (tabEnum) {
+      case tabEnum === libraryTabEnums.PUBLISHED_WORKSPACES:
+        this.reloadPublishedWorkspaces();
+        break;
+      default:
+        break;
+    }
+  }
+
+  async reloadPublishedWorkspaces() {
+    this.setState({workspacesLoading: true});
+    try {
+      const workspacesReceived = await workspacesApi().getPublishedWorkspaces();
+      workspacesReceived.items.sort(
+        (a, b) => a.workspace.name.localeCompare(b.workspace.name));
+      this.setState({workspaceList: workspacesReceived.items
+          .map(w => new WorkspacePermissions(w))});
+      this.setState({workspacesLoading: false});
+    } catch (e) {
+      const response = ErrorHandlingService.convertAPIError(e) as unknown as ErrorResponse;
+      this.setState({errorText: response.message});
+    }
+  }
+
+  render() {
+    const {profile} = this.props.profileState;
```

Comment:
You appear to only use profile email. I think it would be worth deconstructing to just that here? To be explicit on what's being used.